jimp.d.ts

```diff
@@ -30,285 +166,413 @@ declare namespace Jimp {
+            cb: Jimp.ImageCallback<T>
+        ): T;
+        scanQuiet(
             x: number,
             y: number,
-            cb?: Jimp.ImageCallback
-        ): this;
-        brightness(val: number, cb?: Jimp.ImageCallback): this;
-        contrast(val: number, cb?: Jimp.ImageCallback): this;
-        posterize(n: number, cb?: Jimp.ImageCallback): this;
-        histogram(): { r: number[]; g: number[]; b: number[] };
-        normalize(cb?: Jimp.ImageCallback): this;
-        invert(cb?: Jimp.ImageCallback): this;
-        mirror(
-            horizontal: boolean,
-            vertical: boolean,
-            cb?: Jimp.ImageCallback
-        ): this;
-        flip(
-            horizontal: boolean,
-            vertical: boolean,
-            cb?: Jimp.ImageCallback
-        ): this;
-        gaussian(r: number, cb?: Jimp.ImageCallback): this;
-        blur(r: number, cb?: Jimp.ImageCallback): this;
-        convolution(
-            kernel: any,
-            edgeHandling: number | Jimp.ImageCallback,
-            cb?: Jimp.ImageCallback
-        ): this;
-
-        greyscale(cb?: Jimp.ImageCallback): this;
-        grayscale(cb?: Jimp.ImageCallback): this;
-        sepia(cb?: Jimp.ImageCallback): this;
-        opacity(f: any, cb?: any): this;
-        fade(f: any, cb?: any): this;
-        opaque(cb: any): this;
-        resize(
             w: number,
             h: number,
-            mode?: string | Jimp.ImageCallback,
-            cb?: Jimp.ImageCallback
+            f: (this: this, x: number, y: number, idx: number) => any
         ): this;
-        cover(
+        scanQuiet<T>(
+            x: number,
+            y: number,
             w: number,
             h: number,
-            alignBits?: number,
-            mode?: string | Jimp.ImageCallback,
-            cb?: Jimp.ImageCallback
-        ): this;
-        contain(
+            f: (this: this, x: number, y: number, idx: number) => any,
+            cb: Jimp.ImageCallback<T>
+        ): T;
+        crop(x: number, y: number, w: number, h: number): this;
+        crop<T>(
+            x: number,
+            y: number,
             w: number,
             h: number,
-            alignBits?: number,
-            mode?: string | Jimp.ImageCallback,
-            cb?: Jimp.ImageCallback
-        ): this;
-        scale(
-            f: number,
-            mode?: string | Jimp.ImageCallback,
-            cb?: Jimp.ImageCallback
-        ): this;
-        scaleToFit(
+            cb: Jimp.ImageCallback<T>
+        ): T;
+        cropQuiet(x: number, y: number, w: number, h: number): this;
+        cropQuiet<T>(
+            x: number,
+            y: number,
             w: number,
             h: number,
-            mode?: any,
-            cb?: Jimp.ImageCallback
-        ): this;
+            cb: Jimp.ImageCallback<T>
+        ): T;
+
+        // Color methods
+        brightness(val: number): this;
+        brightness<T>(val: number, cb: Jimp.ImageCallback<T>): T;

```

Comment:
is the repetition needed? why doesn't an optional param work?

Reply:
Optional param doesn't work because the two versions have different return types